contracts/lazyMint/LazyMintERC1155.sol

```diff
@@ -0,0 +1,606 @@
+
+{
+    using Strings for uint256;
+
+    /// @dev Only TRANSFER_ROLE holders can have tokens transferred from or to them, during restricted transfers.
+    bytes32 public constant TRANSFER_ROLE = keccak256("TRANSFER_ROLE");
+    /// @dev Only MINTER_ROLE holders can lazy mint NFTs (i.e. can call functions prefixed with `lazyMint`).
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+
+    /// @dev The address interpreted as native token of the chain.
+    address public constant NATIVE_TOKEN = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    /// @dev The address of the native token wrapper contract.
+    address public immutable nativeTokenWrapper;
+
+    /// @dev The adress that receives all primary sales value.
+    address public defaultSaleRecipient;
+
+    /// @dev The next token ID of the NFT to "lazy mint".
+    uint256 public nextTokenIdToMint;
+
+    /// @dev Contract interprets 10_000 as 100%.
+    uint64 private constant MAX_BPS = 10_000;
+
+    /// @dev The % of secondary sales collected as royalties. See EIP 2981.
+    uint64 public royaltyBps;
+
+    /// @dev The % of primary sales collected by the contract as fees.
+    uint120 public feeBps;
+
+    /// @dev Whether transfers on tokens are restricted.
+    bool public transfersRestricted;
+
+    /// @dev Contract level metadata.
+    string public contractURI;
+
+    /// @dev The protocol control center.
+    ProtocolControl internal controlCenter;
+
+    uint[] private baseURIIndices;
+    
+    /// @dev End token Id => URI that overrides `baseURI + tokenId` convention.
+    mapping(uint256 => string) private baseURI;
+    /// @dev Token ID => total circulating supply of tokens with that ID.
+    mapping(uint256 => uint256) public totalSupply;
+    /// @dev Token ID => public mint conditions for tokens with that ID.
+    mapping(uint256 => PublicMintConditions) public mintConditions;
+    /// @dev Token ID => the address of the recipient of primary sales.
+    mapping(uint256 => address) public saleRecipient;
+
+    /// @dev Checks whether caller has DEFAULT_ADMIN_ROLE on the protocol control center.
+    modifier onlyProtocolAdmin() {
+        require(
+            controlCenter.hasRole(controlCenter.DEFAULT_ADMIN_ROLE(), _msgSender()), 
+            "LazyMintERC1155: not protocol admin."
+        );
+        _;
+    }
+
+    /// @dev Checks whether caller has DEFAULT_ADMIN_ROLE.
+    modifier onlyModuleAdmin() {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()), "LazyMintERC1155: not module admin.");
+        _;
+    }
+
+    /// @dev Checks whether caller has MINTER_ROLE.
+    modifier onlyMinter() {
+        require(hasRole(MINTER_ROLE, _msgSender()), "LazyMintERC1155: not minter.");
+        _;
+    }
+
+    constructor(
+        string memory _contractURI,
+        address payable _controlCenter,
+        address _trustedForwarder,
+        address _nativeTokenWrapper,
+        address _saleRecipient
+    ) 
+        ERC1155("")
```

Comment:
In the ERC 1155 implementation, the constructor parameter for the inherited `ERC1155` is supposed to be the `baseURI` for the entire contract. However, since the contract overrides that `baseURI` pattern, the private `_uri` variable set in the inherited `ERC1155` contract's constructor is never referenced in the lifetime of the contract.

As a result, I left this as `""`, and didn't make it configurable - it's an extra constructor parameter to take care of that'll never be referenced in the lifetime of the contract.